asserts/snap_asserts_test.go

```diff
@@ -154,6 +157,44 @@ func (sds *snapDeclSuite) TestDecodeInvalid(c *C) {
 
 }
 
+func (sds *snapDeclSuite) TestDecodePlugs(c *C) {
+	encoded := `type: snap-declaration
+authority-id: canonical
+series: 16
+snap-id: snap-id-1
+snap-name: first
+publisher-id: dev-id1
+plugs:
+  interface1:
+    allow-auto-connection:
+      slot-attributes:
+        a1: /foo/.*
+  interface2:
+    allow-connection:
+      slot-attributes:
+        a2: /foo/.*
```

Comment:
Well, if the base declaration has a blanket deny-installation then we would want an allow-installation in the snap declaration to override that, no? This is how I thought we would handle the hardcoded logic for lxd and docker.